src/typeahead/typeahead.ts

```diff
@@ -182,19 +180,19 @@ export class NgbTypeahead implements ControlValueAccessor,
   }
 
   ngOnInit(): void {
-    const inputValues$ = _do.call(this._valueChanges, value => {
+    const inputValues$ = this._valueChanges.pipe(tap(value => {
       this._inputValueBackup = value;
       if (this.editable) {
         this._onChange(value);
       }
-    });
-    const results$ = letProto.call(inputValues$, this.ngbTypeahead);
-    const processedResults$ = _do.call(results$, () => {
+    }));
+    const results$ = inputValues$.pipe(this.ngbTypeahead);
```

Comment:
why would you keep these intermediate constants? (`results$`, `inputValues$`)
more readable?

Reply:
Thanks for your review!

To be honest, I just changed the usage of old operators to new ones, without changing anything else. But yes, the code is probably more readable this way than by chaining everything. In particular, the call to the ngbTypeahead input is more apparent this way.

Note that I added an additional tiny commit since your review, to make the code compatible with RxJS 6 (I tested the compatibility with a project of mine, and it works fine after this additional commit)